packages/scheduler/lib/models/tasks.integration.test.ts

```diff
@@ -0,0 +1,172 @@
+        const timeout = 1;
+        await createTask({ createdToStartedTimeoutSecs: timeout });
+        await new Promise((resolve) => void setTimeout(resolve, timeout * 1100));
+        const expired = (await tasks.expiresIfTimeout()).unwrap();
+        expect(expired).toHaveLength(1);
+    });
+    it('should expires tasks if startedToCompletedTimeoutSecs is reached', async () => {
+        const timeout = 1;
+        await startTask({ startedToCompletedTimeoutSecs: timeout });
+        await new Promise((resolve) => void setTimeout(resolve, timeout * 1100));
+        const expired = (await tasks.expiresIfTimeout()).unwrap();
+        expect(expired).toHaveLength(1);
+    });
+    it('should expires tasks if heartbeatTimeoutSecs is reached', async () => {
+        const timeout = 1;
+        await startTask({ startedToCompletedTimeoutSecs: timeout });
+        await new Promise((resolve) => void setTimeout(resolve, timeout * 1100));
+        const expired = (await tasks.expiresIfTimeout()).unwrap();
+        expect(expired).toHaveLength(1);
+    });
+});
+
+async function createTaskWithState(state: TaskState): Promise<Task> {
+    switch (state) {
+        case 'CREATED':
+            return createTask();
+        case 'STARTED':
+            return startTask();
+        case 'FAILED':
+            return startTask().then(async (t) => (await tasks.transitionState({ taskId: t.id, newState: 'FAILED' })).unwrap());
+        case 'SUCCEEDED':
+            return startTask().then(async (t) => (await tasks.transitionState({ taskId: t.id, newState: 'SUCCEEDED', output: { foo: 'bar' } })).unwrap());
+        case 'EXPIRED':
+            return startTask().then(async (t) => (await tasks.transitionState({ taskId: t.id, newState: 'EXPIRED' })).unwrap());
+        case 'CANCELLED':
+            return startTask().then(async (t) => (await tasks.transitionState({ taskId: t.id, newState: 'CANCELLED' })).unwrap());
+    }
+}
+
+async function createTask(props?: Partial<tasks.TaskProps>): Promise<Task> {
+    return tasks
+        .create({
+            name: props?.name || rndString(),
+            payload: props?.payload || {},
+            groupKey: props?.groupKey || rndString(),
+            retryMax: props?.retryMax || 3,
+            retryCount: props?.retryCount || 1,
+            startsAfter: props?.startsAfter || new Date(),
+            createdToStartedTimeoutSecs: props?.createdToStartedTimeoutSecs || 10,
+            startedToCompletedTimeoutSecs: props?.startedToCompletedTimeoutSecs || 20,
+            heartbeatTimeoutSecs: props?.heartbeatTimeoutSecs || 5
+        })
+        .then((t) => t.unwrap());
+}
+
+async function startTask(props?: Partial<tasks.TaskProps>): Promise<Task> {
+    return createTask(props).then(async (t) => (await tasks.transitionState({ taskId: t.id, newState: 'STARTED' })).unwrap());
+}
+
+function rndString(): string {
+    return (Math.random() + 1).toString(36).substring(2, 5);
+}
```

Comment:
every functions could be in test.helpers.ts for clarity

Reply:
I'll move then if I ever need them somewhere else